src/js/client.ts

```diff
@@ -1,33 +1,75 @@
+import { BrowserClient, defaultIntegrations, defaultStackParser, makeFetchTransport } from '@sentry/browser';
+import { BrowserTransportOptions } from '@sentry/browser/types/transports/types';
+import { FetchImpl } from '@sentry/browser/types/transports/utils';
 import { BaseClient } from '@sentry/core';
+import { Event, EventHint, SeverityLevel, Transport } from '@sentry/types';
+// @ts-ignore LogBox introduced in RN 0.63
+import { Alert, LogBox, YellowBox } from 'react-native';
 
-import { ReactNativeBackend } from './backend';
-import { ReactNativeOptions } from './options';
+import { ReactNativeClientOptions } from './options';
+import { NativeTransport } from './transports/native';
 import { NATIVE } from './wrapper';
 
 /**
  * The Sentry React Native SDK Client.
  *
- * @see ReactNativeOptions for documentation on configuration options.
+ * @see ReactNativeClientOptions for documentation on configuration options.
  * @see SentryClient for usage documentation.
  */
-export class ReactNativeClient extends BaseClient<
-  ReactNativeBackend,
-  ReactNativeOptions
-> {
+export class ReactNativeClient extends BaseClient<ReactNativeClientOptions> {
+
+  private readonly _browserClient: BrowserClient;
+
   /**
    * Creates a new React Native SDK instance.
    * @param options Configuration options for this SDK.
    */
-  public constructor(options: ReactNativeOptions) {
-    super(ReactNativeBackend, options);
+  public constructor(options: ReactNativeClientOptions) {
+    super(options);
+    // This is a workaround for now using fetch on RN, this is a known issue in react-native and only generates a warning
+    // YellowBox deprecated and replaced with with LogBox in RN 0.63
+    if (LogBox) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      LogBox.ignoreLogs(['Require cycle:']);
+    } else {
+      // eslint-disable-next-line deprecation/deprecation
+      YellowBox.ignoreWarnings(['Require cycle:']);
+    }
+
+    void this._initNativeSdk();
+
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    options.transport = this._chooseTransport;
+    options.stackParser = defaultStackParser;
+
+    this._browserClient = new BrowserClient({
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      transport: this._chooseTransport,
+      stackParser: defaultStackParser,
+      integrations: defaultIntegrations,
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public eventFromException(_exception: unknown, _hint?: EventHint): PromiseLike<Event> {
+    return this._browserClient.eventFromException(_exception, _hint);
```

Comment:
You might be able to ignore my suggestion from above, and remove these overloaded methods by just extended the `BrowserClient` instead of the `BaseClient`.